src/compiler/checker.ts

```diff
@@ -24865,8 +24865,8 @@ namespace ts {
             const declaration = signature.declaration;
             const modifiers = getSelectedModifierFlags(declaration, ModifierFlags.NonPublicAccessibilityModifier);
 
-            // Public constructor is accessible.
-            if (!modifiers) {
+            // (1) Public constructors and (2) constructor functions are always accessible.
+            if (!modifiers || isFunctionLikeDeclaration(declaration)) {

```

Comment:
oops, isFunctionLikeDeclaration includes constructors. I think I'll flip the check and exclude SyntaxKind.Constructor explicitly.